src/main/java/com/google/impactdashboard/server/api_utilities/ResourceRetriever.java

```diff
@@ -75,6 +77,60 @@ protected ProjectListRetriever(CloudResourceManager cloudResourceManagerService)
     return projects;
   }
 
+  /**
+   * Returns the top level ancestor to a project which is the organization the project belongs under.
+   * @param projectId the project the organization id is being retrieved for.
+   */
+  public String getOrganizationId(String projectId){
+    List<Ancestor> ancestors;
+    try {
+       ancestors = getProjectAncestry(projectId);
+    } catch (IOException io) {
+      throw new RuntimeException("Failed to get ancestors: " + io.getMessage());
+    }
+    if(ancestors == null) {
+      return "N/A";
+    }
+    return ancestors.get(ancestors.size()-1).getResourceId().getId();
+  }
+
+  /**
+   * Returns the ancestor hierarchy for the specified project.
+   * @param projectId the project to get the ancestors for
+   */
+  private List<Ancestor> getProjectAncestry(String projectId) throws IOException {
+    CloudResourceManager.Projects.GetAncestry ancestry = cloudResourceManagerService.
+        projects().getAncestry(projectId, new GetAncestryRequest());
+    GetAncestryResponse response = ancestry.execute();
+    return response.getAncestor();
+  }
+
+  /**
+   * Returns the name of the organization that has the id specified.
+   * @param organizationId The id of the organization the name is retrieved for.
+   */
+  public String getOrganizationName(String organizationId) {
+    Organization organization;
+    try {
+      organization = searchOrganizationIds(organizationId);
+    } catch (IOException io) {
+      organization = null;
+//      throw new RuntimeException("Could not find any organization with id:" + organizationId);
```

Comment:
is this the right decision?

If so, please remove the commented out code.

Reply:
Yes there was no other exception that was more specific for my to filter the IOException differently I will remove the comment